src/core/public/workspace/workspaces_service.ts

```diff
@@ -56,6 +57,7 @@ export class WorkspacesService implements CoreService<WorkspacesSetup, Workspace
   private currentWorkspace$ = new BehaviorSubject<WorkspaceObject | null>(null);
   private initialized$ = new BehaviorSubject<boolean>(false);
   private client$ = new BehaviorSubject<IWorkspaceClient | null>(null);
+  private workspaceError$ = new BehaviorSubject<string>('');
```

Comment:
Why we need to add a new `workspaceError$` property? Can we reuse existing `currentWorkspaceId$` and `currentWorkspace$`?

Reply:
Previously I did this but this causes the `WORKSPACE_IS_STALE` error to be throw out and we unable to catch it, so the error can only be caught by the `FatalErrorsService`. In this way, we cannot use the workspace fatal error page anymore. 

In addition, I think use a additional error subject also make sense, so we can handle any workspace specific error any do the error handling only toward this error subject.